internal/p2p/router.go

```diff
@@ -177,30 +194,38 @@ func (o *RouterOptions) Validate() error {
 // quality of service.
 type Router struct {
 	*service.BaseService
-	logger log.Logger
-
-	metrics     *Metrics
-	options     RouterOptions
-	privKey     crypto.PrivKey
-	peerManager *PeerManager
-	chDescs     []*ChannelDescriptor
-	transport   Transport
-	endpoint    *Endpoint
-	connTracker connectionTracker
-
-	peerMtx    sync.RWMutex
-	peerQueues map[types.NodeID]queue // outbound messages per peer for all channels
-	// the channels that the peer queue has open
-	peerChannels     map[types.NodeID]ChannelIDSet
-	queueFactory     func(int) queue
+	logger  log.Logger
+	metrics *Metrics
+
+	options RouterOptions
+	privKey crypto.PrivKey
+	chDescs []*ChannelDescriptor
+
 	nodeInfoProducer func() *types.NodeInfo
 
-	// FIXME: We don't strictly need to use a mutex for this if we seal the
-	// channels on router start. This depends on whether we want to allow
-	// dynamic channels in the future.
-	channelMtx      sync.RWMutex
-	channelQueues   map[ChannelID]queue // inbound messages from all peers to a single channel
-	channelMessages map[ChannelID]proto.Message
+	legacy struct {
```

Comment:
I suppose this is likely to shrink over time, but when there are this many fields i'd recommend defining a type. The rationale is simply pragmatic: Initializing the fields is a one-by-one thing unless you restate the type, which can be tedious to read.